types/orchestrator/index.d.ts

```diff
@@ -1,107 +1,141 @@
-// Type definitions for Orchestrator
+// Type definitions for Orchestrator 0.3
 // Project: https://github.com/orchestrator/orchestrator
-// Definitions by: Qubo <https://github.com/tkQubo>
+// Definitions by: Qubo <https://github.com/tkQubo>, TeamworkGuy2 <https://github.com/TeamworkGuy2>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+/// <reference types="node" />
+
+import * as events from "events";
+import * as stream from "stream";
 import * as Q from "q";
 
-declare type Strings = string|string[];
+type _Sequencify = (tasks: Array<{ dep: string[]; }>, names: string[]) => { sequence: string[]; missingTasks: string[]; recursiveDependencies: string[]; };
+
+type _runTask = (task: Orchestrator.TaskFunc, done: (err: any, meta: Orchestrator.Meta) => void) => void;
+
+type Strings = string|string[];
+
+/** A module for sequencing and executing tasks and dependencies in maximum concurrency
+ */
+declare class Orchestrator extends events.EventEmitter {
+    doneCallback?: (error?: any) => any;
+    isRunning: boolean;
+    seq: any[];
+    tasks: { [name: string]: Orchestrator.Task };
+
+    reset(): Orchestrator;
+
+    /** Define a task
+     * @param name The name of the task.
+     * @param fn The function that performs the task's operations. For asynchronous tasks, you need to provide a hint when the task is complete:
+     *  - Take in a callback
+     *  - Return a stream or a promise
+     */
+    add(name: string, fn?: Orchestrator.TaskFunc): Orchestrator;
```

Comment:
If you want to stop using call signature `AddMethod`, you can remove its definition